### DISCUSSION TITLE
B cell lineage tree

### DISCUSSION THREAD

**Discussion by Januaryyiyue** [milaboratory/mixcr#1420] [Q&A] [answered]
Is it possible to do B cell lineage analysis but not use `analyze`? For my custom analysis I am running `align`, `assemblePartial`, `extend` and `exportClones`.

If so, which starting files from my custom analysis should I use to build the tree? Or is it better to somehow incorporate all these steps into `analyze`?

Thank you

**Comment by mizraelson** [comment]
What kind of data do you have? RNA-seq?

**Comment by Januaryyiyue** [comment]
Hi,

Thank you for your prompt response.

I am using the RNA-seq preset. The method I used is a targeted-sequencing technique (CapIG-seq) that captures the 3' of the V region, the CDR3 region and the 5' of the J region.

Best
________________________________

**Reply by mizraelson** [reply, marked as answer]
Since this protocol utilizes DNA (as indicated in the publication I have found, correct?), the appropriate command should be:

```
mixcr analyze exome-seq \
    --species hsa \
      input_R1.fastq.gz \
      input_R2.fastq.gz \
      result
```

This preset includes the steps `align`, `assemblePartial`, `extend`, `assembleContigs`, and `exportClones`. Please note that `extend` is automatically used for only TCR data. We can't apply it for BCR CDR3 extension due to potential hypermutations.

Utilizing this data for tree reconstruction will be quite challenging as it doesn't cover the full length of V and J genes comprehensively. Nevertheless, tree reconstruction necessitates equal coverage of all clones (i.e., the same gene segments must be covered). To filter clones based on this criteria, you can use the `--assemble-contigs-by {FR3Begin:FR4End}` parameter. Be sure to adjust the gene feature to match the coverage in your data.

Let me know if you need any further clarification.
